packages/fx-core/src/component/driver/oauth/utility/utility.ts

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { ProjectType, SpecParser } from "@microsoft/m365-spec-parser";
+import { getAbsolutePath } from "../../../utils/common";
+import { DriverContext } from "../../interface/commonArgs";
+import { CreateOauthArgs } from "../interface/createOauthArgs";
+import { isCopilotAuthEnabled } from "../../../../common/featureFlags";
+import { OpenAPIV3 } from "openapi-types";
+import { isEqual } from "lodash";
+import { maxDomainPerApiKey } from "./constants";
+import { OauthDomainInvalidError } from "../error/oauthDomainInvalid";
+import { OauthFailedToGetDomainError } from "../error/oauthFailedToGetDomain";
+import { OauthAuthInfoInvalid } from "../error/oauthAuthInfoInvalid";
+
+export interface OauthInfo {
+  domain: string[];
+  authorizationEndpoint: string;
+  tokenExchangeEndpoint: string;
+  tokenRefreshEndpoint?: string;
+  scopes: string[];
+}
+
+interface AuthInfo {
+  authorizationUrl: string;
+  tokenUrl: string;
+  refreshUrl?: string;
+  scopes: string[];
+}
+
+export async function getandValidateOauthInfoFromSpec(
+  args: CreateOauthArgs,
+  context: DriverContext,
+  actionName: string
+): Promise<OauthInfo> {
+  const absolutePath = getAbsolutePath(args.apiSpecPath, context.projectPath);
+  const parser = new SpecParser(absolutePath, {
+    allowAPIKeyAuth: false,
+    allowBearerTokenAuth: isCopilotAuthEnabled(),
```

Comment:
suggest to use the new API to access feature flag: `featureFlagManager.getBooleanValue(FeatureFlags.CopilotAuth)`

Reply:
Will fix in another pr. Thanks!